service/src/main/java/bio/terra/profile/app/controller/AzureApiController.java

```diff
@@ -18,24 +21,38 @@ public class AzureApiController implements AzureApi {
   private final HttpServletRequest request;
   private final AzureService azureService;
   private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final PolicyServiceConfiguration policyServiceConfiguration;
 
   @Autowired
   public AzureApiController(
       HttpServletRequest request,
       AzureService azureService,
-      AuthenticatedUserRequestFactory authenticatedUserRequestFactory) {
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      PolicyServiceConfiguration policyServiceConfiguration) {
     this.request = request;
     this.azureService = azureService;
     this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.policyServiceConfiguration = policyServiceConfiguration;
   }
 
   @Override
   public ResponseEntity<AzureManagedAppsResponseModel> getManagedAppDeployments(
       UUID azureSubscriptionId, Boolean includeAssignedApplications) {
     final AuthenticatedUserRequest userRequest = authenticatedUserRequestFactory.from(request);
-    var managedApps =
-        this.azureService.getAuthorizedManagedAppDeployments(
-            azureSubscriptionId, includeAssignedApplications, userRequest);
+    List<AzureManagedAppModel> managedApps;
+
+    // check azure control plane feature flag
+    if (this.policyServiceConfiguration.getAzureControlPlaneEnabled()) {
+      // get ServiceCatalog deployed managed apps
+      managedApps =
+          this.azureService.getServiceCatalogManagedAppDeployments(
+              azureSubscriptionId, includeAssignedApplications, userRequest);
+    } else {
+      // get Marketplace deployed managed apps
+      managedApps =
+          this.azureService.getAuthorizedManagedAppDeployments(
+              azureSubscriptionId, includeAssignedApplications, userRequest);
+    }
```

Comment:
I am wondering if this would be simpler if it diverged in `AzureService`. `getServiceCatalogManagedAppDeployments` and `getAuthorizedManagedAppDeployments` are almost identical methods, the only difference that I'm seeing is one filters based on `isServiceCatalogTerraManagedApp` and the other uses `isAuthedTerraManagedApp`

Reply:
We can push the switch down into AzureService isAuthedTerraManagedApp(). I had it there initially, but thought this way brings the feature flag up to a level that shows we have different behavior for a single API. Take a look at AzureService in cancelled PR #415 and let me know if you prefer that approach

Reply:
I agree the controller is not the right place. Controllers should have little to no business logic.